### DISCUSSION TITLE
3.6 pre-release changes

### DISCUSSION THREAD

**Discussion by kjk** [sumatrapdfreader/sumatrapdf#3821] [Announcements]
This is a log of changes in 3.6 pre-release. They'll eventually be released as 3.6 but if you want to try them sooner, you can install [pre-release version](https://www.sumatrapdfreader.org/prerelease).

**Oct 26 2023**

Updated [mupdf](https://mupdf.com/). We use mupdf for PDF / epub / XPS rendering and paused updating it while finishing 3.5 release.

The update fixes some PDF documents.

**Comment by kjk** [comment]
**Oct 26 2023**

Make loading from command palette async so that it doesn't block the UI if the document is slow to load (e.g. a large epub files).

**Reply by stanantos** [reply]
Witaj Krzysztofie Kowalczyku, mam do Ciebie wielką prośbę. Używam Twojego pliku Sumatra PDF od dłuższego czasu, jest świetny! Niestety ostatnią wersją w której poszedł na zmianę języka na czeski była wersja 3.4.6 - od wersji 3.5 (3.5.1, 3.5.2, prerel.3.6...) to w ogóle nie działa. W ten sam sposób drugi najbliższy język słowacki - słowacki - nie działa dla nas. Jednocześnie polski, rosyjski i domyślny angielski działają bez problemów, a jeśli wybiorę czeski lub słowacki, odpowiedni plik *.txt w linii interfejsu użytkownika jest modyfikowany na cz lub sk, ale język się nie przełącza i zwykle pozostaje tylko angielski. Bardzo mi przykro. Nie działa w żadnym z instalatorów (32 + 64-bitowy) ani w żadnej z dwóch przenośnych wersji programu.
Byłbym bardzo wdzięczny za pomoc.
Miłego dnia Stanislav Antoš

[EDGE translation !]
Hello Krzysztof Kowalczyk, I have a big request for you. I've been using your Sumatra PDF for quite some time now, it's great! Unfortunately the last version in which he switched to Czech was version 3.4.6 - since version 3.5 (3.5.1, 3.5.2, prerel.3.6...) it doesn't work at all. In the same way, the second closest Slovak language - Slovak - does not work for us. At the same time, Polish, Russian and default English work without problems, and if I choose Czech or Slovak, the corresponding *.txt file in the UI line is modified to cz or sk, but the language does not switch and usually only English remains. I'm so sorry. It does not work in any of the installers (32 + 64-bit) or in any of the two portable versions of the program.
I would be very grateful for your help.
Have a nice day Stanislav Antoš

**Reply by GitHubRulesOK** [reply]
@stanantos 
please see https://github.com/sumatrapdfreader/sumatrapdf/issues/3867

**Reply by stanantos** [reply]
Bardzo dziękujemy za przesłany link z tym samym pytaniem i odpowiedzią na nie, z niecierpliwością czekamy, aż wersja 3.5.3 rozwiąże problemy z przejściem na język czeski (słowacki).
Ładny dzień Stanislav Antoš
Thank you very much for the sent link with the same question and the answer to it, we will look forward to the fact that version 3.5.3 will solve the problems with switching to Czech (Slovak).
Nice day  Stansialv Antoš

**Reply by toquanghiep** [reply]
I have referred to version 3.6 but it still does not support Vietnamese Unicode
In this case I used Times New Roman font
[Example.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/14069191/Example.pdf)
<img width="842" alt="not use sumatra" src="https://github.com/sumatrapdfreader/sumatrapdf/assets/12733827/007f3a1a-db65-402d-9938-a90e224d2b46">
<img width="656" alt="use sumatra" src="https://github.com/sumatrapdfreader/sumatrapdf/assets/12733827/479203cd-bea7-4b61-ae86-a09a33884c19">

**Reply by GitHubRulesOK** [reply]
SumatraPDF is based on MuPDF and if the font is not Included it reverts to the stock 256 character Times and Helvetica (Arial) Which dont include Vietnamese characters. If the fonts are sub-set (included) for archival use (ISO standard PDF/A) then they will be used and the file is bigger.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/ddabb75c-f12e-4c49-8e15-64166720fe91)

**Comment by TheSQLGuru** [comment]
Am I to take it that the only work being done on this next release (so far) is mupdf?? If not, is there another location I can find all items being worked on for this next release?

**Reply by GitHubRulesOK** [reply]
3.5.3 IF released would be a minor maintenance release those items that are directly affected are already stacked so 14 of about 2 dozen are translation updates, and affect the language percentage of supported, but not all single words would be automatically elevated into usage.
If not released they are already in 3.6 pre-release which is in progress on a loosely daily basis  (but if no change from day before it is not uploaded.)
3.6 pre-release is currently about 150 changes more advanced than 3.5.3 candidates.
and many commits may go to and fro so the list will be published as a summary when done. MuPDF may be update dozens of time  but will simply be one overarching comment, much like "minor bug fixes"

**Comment by Erwan62** [comment]
Hello again
I'm currently proofreading a working document. I really like Sumatra's diversified annotation system. However, I'm confused about the "free text" because there's no size management; I was expecting to be able to edit it by dragging the borders.
Have a nice day!

**Comment by fbrazdik** [comment]
What is the time horizon for releasing 3.6 as stable version?

**Reply by GitHubRulesOK** [reply]
@fbrazdik 
Never has been a "timeframe" look at history of official releases
They have varied from one day to several (or more) years

**Reply by GitHubRulesOK** [reply]
@fbrazdik the 3.6 release time is (some time ago now) "today"
